include/affinity.h

```diff
@@ -0,0 +1,14 @@
+#pragma once
+
+#include <cstdint>
+
+namespace celerity {
+namespace detail {
+
+	uint32_t affinity_cores_available();
+
+	// a priori we need 3 threads plus 1 for workers. This depends on the application invoking celerity.
```

Comment:
Could we rephrase this to "plus at least 1 for host-task workers`? Because that's why it's application dependent. Also this does not yet factor in CUDA/MPI/whatnot. (Maybe we should bump this to at least 5?).